### PR TITLE
Get rid of sshd in the traffic-agent

### PR DESCRIPTION
**Pull request by thallgren**
## Description

This PR contains what is needed to get rid of sshd in the traffic-agent and all use of ssh in Telepresence. It sits on top of #1685.

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [x] My change is adequately tested.

### CODE REVIEW

File: cmd/traffic/cmd/agent/agent.go [hunk 1]
```diff
@@ -128,15 +132,9 @@ func Main(ctx context.Context, args ...string) error {
 	}
 	dlog.Infof(ctx, "%+v", config)
 
-	hostname, err := os.Hostname()
-	if err != nil {
-		dlog.Infof(ctx, "hostname: %+v", err)
-		hostname = fmt.Sprintf("unknown: %+v", err)
-	}
-
 	info := &rpc.AgentInfo{
```
Comment (donnyyung) [inline cmd/traffic/cmd/agent/agent.go:135]: Is there a PR for changes to the proprietary agent for this change set?  I noticed when I ran the smoke tests, and tried to create an intercept with a preview URL I got this error:
```
telepresence: error: creating preview domain: rpc error: code = Unavailable desc = transport is closing
rpc error: code = Unavailable desc = connection closed
```

Similarly, I found this in the traffic-agent logs: 
```
time="2021-05-04 13:36:52" level=info msg="talking to manager:rpc error: code = Canceled desc = context canceled" func=main.sidecar file="github.com/datawire/telepresence2/proprietary/cmd/app-sidecar/sidecar.go:246" PID=1 THREAD=/sidecar
```
One of the arguments to TalkToManager is the AgentInfo so I suspect this change needs to be made in the proprietary agent repo as well.

File: base-image/Dockerfile [hunk 2]
```diff
@@ -12,26 +12,12 @@
...
-    -e 's/^AllowTcpForwarding no/AllowTcpForwarding yes/' \
-    -e 's/^GatewayPorts no/GatewayPorts yes/' \
-    -e '$aClientAliveInterval 1' \
-    -e '$aPermitEmptyPasswords yes' \
-    -e '$aPort 8022' \
-    -e '$aClientAliveCountMax 10' \
-    -e '$aPermitRootLogin yes' \
-    /etc/ssh/sshd_config && \
-  chmod -R g+r /etc/ssh && \
+  apk add --no-cache openssh-sftp-server && \
```
Comment (LukeShu) [inline base-image/Dockerfile]: Don't remove `ca-certificates`, we still need it for validating Ambassador Cloud's TLS cert.

File: base-image/Dockerfile [hunk 3]
```diff
@@ -12,26 +12,12 @@
...
   mkdir /home/telepresence && \
```
Comment (LukeShu) [inline base-image/Dockerfile:18]: Can we adjust the license-key mountpoint, and get rid of `/home/telepresence` too?

**Review by LukeShu — APPROVED** [review]
1. Don't remove the `ca-certificates` package.
2. It'll need rebased because of the other helm chart commits in the preceding PR.

Otherise, LGTM.

**Review by LukeShu — CHANGES_REQUESTED** [review]
[oops, clicked the wrong thing]

File: base-image/Dockerfile (hunk 3 above)
Comment (thallgren) [reply base-image/Dockerfile:18]: Where should it go? We need to mount it from somewhere.

Comment (LukeShu) [reply base-image/Dockerfile:18]: `/tel_license`?

But that doesn't need to go in this PR if you don't want.

**Review by LukeShu — APPROVED** [review]

**Pull request merged**